docs/proposals/14-reversed-cluster-vpn.md

```diff
@@ -0,0 +1,276 @@
+
+The solution is depicted in this diagram:
+
+![alt text](assets/CurrentClusterVPN.png "Overview Current Cluster VPN")
+
+These are the essential parts of the OpenVPN client configuration in the `vpn-seed` sidecar container:
+```
+# use TCP instead of UDP (commonly not supported by load balancers)
+proto tcp-client
+
+[...]
+
+# get all routing information from server
+pull
+
+tls-client
+key "/srv/secrets/vpn-seed/tls.key"
+cert "/srv/secrets/vpn-seed/tls.crt"
+ca "/srv/secrets/vpn-seed/ca.crt"
+
+tls-auth "/srv/secrets/tlsauth/vpn.tlsauth" 1
+cipher AES-256-CBC
+
+# https://openvpn.net/index.php/open-source/documentation/howto.html#mitm
+remote-cert-tls server
+
+# pull filter
+pull-filter accept "route 100.64.0.0 255.248.0.0"
+pull-filter accept "route 100.96.0.0 255.224.0.0"
+pull-filter accept "route 10.1.60.0 255.255.252.0"
+pull-filter accept "route 192.168.123."
+pull-filter ignore "route"
+pull-filter ignore redirect-gateway
+pull-filter ignore route-ipv6
+pull-filter ignore redirect-gateway-ipv6
+```
+
+Encryption is based on SSL certificates with an additional HMAC signature to all SSL/TLS handshake packets. As multiple clients connect to the OpenVPN server in the shoot cluster, all clients must be assigned a unique IP address. This is done by the OpenVPN server pushing that configuration to the client (keyword `pull`). As this is potentially problematic because the OpenVPN server runs in an untrusted environment there are pull filters denying all but IP address allocation and route distribution.
+
+The OpenVPN server running in the shoot cluster is configured as follows:
+
+```
+mode server
+tls-server
+proto tcp4-server
+dev tun0
+
+[...]
+
+server 192.168.123.0 255.255.255.0
+
+push "route 10.243.0.0 255.255.128.0"
+push "route 10.243.128.0 255.255.128.0"
+
+duplicate-cn
+
+key "/srv/secrets/vpn-shoot/tls.key"
+cert "/srv/secrets/vpn-shoot/tls.crt"
+ca "/srv/secrets/vpn-shoot/ca.crt"
+dh "/srv/secrets/dh/dh2048.pem"
+
+tls-auth "/srv/secrets/tlsauth/vpn.tlsauth" 0
+push "route 10.242.0.0 255.255.0.0"
+```
+
+It is a TCP TLS server and configured to automatically assign IP addresses for OpenVPN clients (`server` directive). In addition, it pushes the shoot cluster node-, pod-, and service ranges to the clients running in the seed cluster (`push` directive). 
+
+**Note:** The network mesh spanned by OpenVPN uses the network range `192.168.123.0 - 192.168.123.255`. This network range cannot be used in either shoot-, or seed clusters. If it is used this might cause subtle problem due to network range overlaps. Unfortunately, this appears not to be well documented but this restriction exists since the very beginning. We should clean up this technical debt as part of the excercise.
```

Comment:
```suggestion
**Note:** The network mesh spanned by OpenVPN uses the network range `192.168.123.0 - 192.168.123.255`. This network range cannot be used in either shoot-, or seed clusters. If it is used this might cause subtle problem due to network range overlaps. Unfortunately, this appears not to be well documented but this restriction exists since the very beginning. We should clean up this technical debt as part of the exercise.
```

Reply:
All issues are resolved with the exception of the HA discussion. From our point of view this should not block this GEP.

Reply:
all earlier PR comments resolved.